crates/services/txpool/src/containers/dependency.rs

```diff
@@ -424,7 +425,7 @@ impl Dependency {
                         *utxo_id,
                         CoinState {
                             is_spend_by: Some(tx.id() as TxId),
-                            depth: max_depth - 1,
+                            depth: max_depth.saturating_sub(1),
```

Comment:
Should we just throw an error here since this function returns a result anyways?

Reply:
We could, but `max_depth` above is always more than zero, so it should be impossible to overflow. I thought about using `checked_sub + expect` here to highlight it, but decided to go with saturating because depth less than zero also doesn't make any sense.

But if you think that `checked_sub + expect` is better,  I can move to it=)